m4a-dec.c

```diff
@@ -121,19 +120,24 @@ int main(int argc, char *argv[]) {
 			}
 			frame_size = info->frameSize * info->numChannels;
 			// Note, this probably doesn't return channels > 2 in the right order for wav
-			wav = wav_write_open(outfile, info->sampleRate, 16, info->numChannels);
+			wav = wav_write_open(outfile, info->sampleRate, sizeof(INT_PCM)*8, info->numChannels);
 			if (!wav) {
 				perror(outfile);
 				status = 1;
 				break;
 			}
 		}
-		for (i = 0; i < frame_size; i++) {
-			uint8_t* out = &output_buf[2*i];
+		for (int i = 0; i < frame_size; i++) {
```

Comment:
This file was written in C89 style (for whatever reasons that was, originally), so at least avoid mixing such changes (like moving "int i" to the for loop) here where they aren't needed.